### PR TITLE
Deactivation popup

### PR DESCRIPTION
**Pull request by pereirinha**
This PR introduces a feedback pop up when the plugin is to be deactivated

### CODE REVIEW

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/deactivate.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+			});
+		});
+
+		// Add event listener to submit the feedbacl.
+		context.submitButton.addEventListener( 'click', function( ev ) {
+			wp.ajax.send( {
+				url: cloudinaryApi.endpoint,
+				data: {
+					reason: context.reason,
+					more: context.more.value,
```
Comment (DavidCramer) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/deactivate.js:64]: @pereirinha - can't read value of null when an option without textarea selected, which stops submission.

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/deactivate.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		// Add event listener to update reason and more container.
+		[ ...context.options ].forEach( ( option ) => {
+			option.addEventListener( 'change', function( ev ) {
+				context.submitButton.removeAttribute( 'disabled' );
+				context.reason = ev.target.value;
+				context.more = ev.target.parentNode.querySelector( 'textarea' );
+			});
+		});
+
+		// Add event listener to submit the feedbacl.
```
Comment (DavidCramer) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/deactivate.js:58]: @pereirinha slight typo `feedbacl`

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/deactivate.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	reason: '',
+	// The more details .
+	more: null,
+	// The deactivation link for the plugin.
+	deactivationUrl: '',
+
+	addEvents: function() {
+		const context = this;
+
+		// Add event listener to deativation links to add the pop up.
```
Comment (DavidCramer) [inline cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/deactivate.js:24]: @pereirinha slight typo `deativation`

**Review by DavidCramer — CHANGES_REQUESTED** [review]

**Comment by dugajean** [comment]
Code looks good. Test pending.

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/deactivate.js (hunk 1 above)
Comment (pereirinha) [reply cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/deactivate.js]: @DavidCramer 

Good catch. This was addressed.

File: cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/deactivate.js (hunk 2 above)
Comment (pereirinha) [reply cloudinary-image-management-and-manipulation-in-the-cloud-cdn/js/src/deactivate.js]: 👍

**Review by DavidCramer — APPROVED** [review]
Looks good and works as expected.

**Pull request merged**